src/session/session_api.c

```diff
@@ -2219,6 +2219,38 @@ __session_get_last_error(WT_SESSION *wt_session, int *err, int *sub_level_err, c
     *err_msg = session->err_info.err_msg;
 }
 
+/*
+ * __session_set_last_error --
+ *     WT_SESSION->set_last_error method.
+ */
+static int
+__session_set_last_error(WT_SESSION *wt_session, int err, int sub_level_err)
```

Comment:
We should add WT_ASSERT calls to check that the caller hasn't mixed up err and sub_level_err. 
I suggest creating a function that returns true if an integer value is a valid sub level error code and false otherwise. Then assert that the sub_level_err is valid, and that err is not a sub level error code.

Reply:
Done in [Added assert to check for valid sub_level_err, formatting changes](https://github.com/wiredtiger/wiredtiger/pull/11350/commits/f0f443ad444cf6fe5f2fa02603569cd679c53dd6)